Doc: OpenFileGDB: document OPENFILEGDB_DEFAULT_STRING_WIDTH configuration option (fixes #9165)

rouault opened the pull request.

elpaso commented on ogr/ogrsf_frmts/openfilegdb/ogropenfilegdblayer.cpp:
```diff
@@ -664,7 +664,7 @@ int OGROpenFileGDBLayer::BuildLayerDefinition()
         // string width is 0, we pick up 65536 by default to mean unlimited
```
this comment needs updates, maybe
```suggestion
        // string width is 0, if OPENFILEGDB_DEFAULT_STRING_WIDTH is not set we pick up 65536 by default to mean unlimited
```

elpaso commented:
I found it confusing that the C++ constant OPENFILEGDB_DEFAULT_STRING_WIDTH and the open option have the same name: if I read this right the first is the hard limit (and the default) and the second is a user-defined default which still needs to be less than the hard limit.

rouault commented:
> I find it confusing that the C++ constant OPENFILEGDB_DEFAULT_STRING_WIDTH

yes, I agree that's confusing. Renamed in an extra commit where I've also added extensive comments to explain the trick

rouault commented:
(note: CIFuzz failure is unrelated, and will be addressed per https://github.com/OSGeo/gdal/pull/9181)
